gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java

```diff
@@ -120,9 +155,75 @@ private void resetMetrics(Map<String, AtomicLong> metrics) {
     }
   }
 
+  /**
+   * Updating the required gcs specific statistics based on httpresponse.
+   *
+   * @param statusCode
+   */
+  private void updateGcsIOSpecificStatistics(int statusCode) {
+
+    if (statusCode == 429) {
+      incrementRateLimitingCounter();
+    }
+
+    if (200 <= statusCode && statusCode < 300) {
+      incrementGcsSuccessCodeCount();
+    }
+
+    if (statusCode >= 400 && statusCode < 500) {
+      incrementGcsClientSideCounter();
+    }
+
+    if (statusCode >= 500 && statusCode < 600) {
+      incrementGcsServerSideCounter();
+    }
+  }
+
+
+  /**
+   * Updating the required gcs specific statistics based on GoogleJsonResponseException.
+   *
+   * @param responseException contains statusCode based on which metrics are updated
+   */
+  @Subscribe
+  private void subsriberOnGoogleJsonResponseException(
+      GoogleJsonResponseException responseException) {
+    updateGcsIOSpecificStatistics(responseException.getStatusCode());
```

Comment:
Whould we check for responseException null? Also can getStatuCode be null?

Reply:
Since the usage of this Subscriber method makes sure that responseException is never null, hence skipping the check. I do not think that getStatusCode() can return null but still added check.